### PR TITLE
Theme Color updates all layers/styles

### PR DESCRIPTION
**Pull request by milg15**
Solves #33.

I am leaving it as a draft as I mentioned in the issue because I don't think the implementation it's optimal.

### CODE REVIEW

**Review by dabbott — CHANGES_REQUESTED** [review]
It's true that this will be slow, but the faster approaches I can think of would be a lot more complex:

- We could maintain a list of index paths of which layers use which theme swatches, but this would be a hassle to keep in sync
- We could maintain a list of references to all color objects, so that we could easily iterate through and modify the ones that are linked to swatches, but then we'd need a way to force a react re-render

I think we can keep the slow approach for now, and revisit in the future

File: packages/noya-state/src/reducers/application.ts [hunk 1]
```diff
@@ -903,14 +915,43 @@ export function reducer(
...
+          const changeColor = (c: Sketch.Color) => {
+            if (!(c.swatchID === swatch.do_objectID)) return;
+
+            c.alpha = color.alpha;
+            c.red = color.red;
+            c.blue = color.blue;
+            c.green = color.green;
+          };
+
+          layerIndexPaths.forEach((layerPath) =>
```
[the remark this answers is not in the document]
Comment (milg15) [reply packages/noya-state/src/reducers/application.ts:942]: oh awesome! I was wondering how to do that

File: packages/noya-state/src/reducers/application.ts [hunk 2]
```diff
@@ -903,14 +915,43 @@ export function reducer(
...
         sharedSwatches.forEach((swatch: Sketch.Swatch) => {
-          if (ids.includes(swatch.do_objectID)) {
-            swatch.value = color;
-          }
+          if (!ids.includes(swatch.do_objectID)) return;
+
+          swatch.value = color;
+
+          const changeColor = (c: Sketch.Color) => {
+            if (!(c.swatchID === swatch.do_objectID)) return;
```
[the remark this answers is not in the document]
Comment (milg15) [reply packages/noya-state/src/reducers/application.ts:934]: This one was definitely a brain lapse😆

**Review by dabbott — APPROVED** [review]

**Comment by milg15** [comment]
Yay!

**Pull request merged**